app/grandchallenge/evaluation/templates/evaluation/submission_form.html

```diff
@@ -84,7 +84,11 @@ <h3>Create a new submission</h3>
             {% include 'evaluation/partials/algorithm_interfaces_information.html' with phase=phase form=form %}
         {% endif %}
 
-        {% crispy form %}
+        <form method="POST">
+            {% csrf_token %}
+            {{ form|crispy }}
+            <input type="submit" name="save" value="Submit" class="btn btn-primary">
+        </form>
```

Comment:
Shouldn't all of the form setup, csrf token and button be handled by crispy forms?

Reply:
I thought the same, but couldn't get it to work. The additional inputs (which are added to form.fields on form initialization) did not render when using `{% crispy form %}`. I must admit that I was a bit lazy and just copied the implementation from the algorithm form (where we also have dynamic form fields). If you have a hunch why the crispy form filter isn't working for those fields, I'm all ears.

Reply:
Without looking at it could it be that the helper is added before the fields are present?

Reply:
That was it! Thanks! I will update the algorithm form as well.

Reply:
Wow, I will buy a lottery ticket on the way home.